ghci-wrapper/src/Language/Haskell/GhciWrapper.hs

```diff
@@ -102,7 +102,7 @@ putExpression :: Interpreter -> Bool -> String -> IO ()
 putExpression Interpreter{hIn = stdin} preserveIt e = do
   hPutStrLn stdin e
   when preserveIt $ hPutStrLn stdin $ "let " ++ itMarker ++ " = it"
-  hPutStrLn stdin marker
+  hPutStrLn stdin (marker ++ " :: String")
```

Comment:
Adding the type signature prevents warnings with `-XOverloadedStrings`, but it has the downside that `String` needs to be in scope.